melange-core/c-lexer-cpp.dylan

```diff
@@ -130,7 +131,11 @@ define function get-macro-params
       <rparen-token> => paren-count := paren-count - 1;
       otherwise => #f;
     end select;
+    empty-token := #f
   finally
+    if (empty-token) // Insert an empty token.
+      list := pair(make(<identifier-token>, string: "", generator: state), list)
+    end if;
```

Comment:
Maybe I'm missing something ... it looks like this is just adding a single empty token to the ``list`` at head of the list (since the list is in reverse order). Why is that the right solution? Shouldn't there be an empty token in the right spot in the list?